platform/viewer/src/appExtensions/GenericRoutesModule/routesModule.js

```diff
@@ -0,0 +1,75 @@
+import asyncComponent from '../../components/AsyncComponent.js';
+
+// Dynamic Import Routes (CodeSplitting)
+const IHEInvokeImageDisplay = asyncComponent(() =>
+  import(
+    /* webpackChunkName: "IHEInvokeImageDisplay" */ '../../routes/IHEInvokeImageDisplay.js'
+  )
+);
+const ViewerRouting = asyncComponent(() =>
+  import(
+    /* webpackChunkName: "ViewerRouting" */ '../../routes/ViewerRouting.js'
+  )
+);
+
+const StudyListRouting = asyncComponent(() =>
+  import(
+    /* webpackChunkName: "StudyListRouting" */ '../../studylist/StudyListRouting.js'
+  )
+);
+const StandaloneRouting = asyncComponent(() =>
+  import(
+    /* webpackChunkName: "ConnectedStandaloneRouting" */ '../../connectedComponents/ConnectedStandaloneRouting.js'
+  )
+);
+const ViewerLocalFileData = asyncComponent(() =>
+  import(
+    /* webpackChunkName: "ViewerLocalFileData" */ '../../connectedComponents/ViewerLocalFileData.js'
+  )
+);
+
+const ROUTES_DEF = {
+  viewer: {
+    template: 'Viewer',
+    component: ViewerRouting,
+  },
+  standAloneViewer: {
+    template: 'StandAloneViewer',
+    component: StandaloneRouting,
+  },
+  studyList: {
+    template: 'StudyList',
+    component: StudyListRouting,
+    condition: appConfig => {
+      return appConfig.showStudyList !== undefined
+        ? appConfig.showStudyList
+        : true;
+    },
+  },
+  local: {
+    template: 'Local',
+    component: ViewerLocalFileData,
+  },
+  IHEInvokeImageDisplay: {
+    template: 'IHEInvokeImageDisplay',
+  },
+  gcloudViewer: {
+    template: 'GCloudViewer',
+    component: ViewerRouting,
+    condition: appConfig => {
+      return !!appConfig.enableGoogleCloudAdapter;
+    },
+  },
+  gcloudStudyList: {
+    template: 'GCloudStudyList',
+    component: StudyListRouting,
+    condition: appConfig => {
+      const showList =
+        appConfig.showStudyList !== undefined ? appConfig.showStudyList : true;
+
+      return showList && !!appConfig.enableGoogleCloudAdapter;
+    },
+  },
+};
+
+export default ROUTES_DEF;
```

Comment:
The goal here being for the `routeModule` to add new "named" route templates that can be used when specifying route definitions in the app level config?

I think we can make this an array of objects where the template name is used. I'm worried the object key values would just add confusion.

Are we thinking:
```js
{
  template: string,
  component: ReactComponent (top level container component)
  condition: ???
}
```

What should condition expect to receive? When/why would we use it?
For IHEInvokedImageDisplay, how/where does that template find it's component?

As we're not specifying routes here, should this actually be a templateModule, or routeTemplateModule? Is the name clear enough for what we're trying to accomplish?

Reply:
- IHEInvokedImageDisplay - revisited and set it back thank you
- Changed name to routeTemplateModule
- Changed to array structure (using template string as identifier)

Reply:
- Changed the name
- Set back IHEInvokedImageDisplay
- Changed to array approach instead of object (using template as identifier)
- Changed name to routeTemplateModule